### ISSUE TITLE
Migrate providerConfig

### ISSUE THREAD

**Issue by davidewatson** [samsung-cnct/k2#269]
# Boilerplate

The k2recon project involves creating a versioned schema to validate configuration files. As a result, some of configuration properties have been renamed, and stanzas have been reorganized. For backwards compatibility while we migrate to the new config, the old stanzas have been left in place alongside the new stanzas.

# providerConfig stanza

This ticket is to remove the old stanza, ```"#/deployment/providerConfig"```, from ansible/roles/kraken.config/files/config.yaml  and update the related Ansible tasks to match. When doing this the k2 schema, ansible/roles/kraken.config/files/schema.json , should be updated as well.

**Comment by davidewatson** [comment]
# Notes

If GKE involves significantly more work, then consider opening another ticket.

The providerConfig stanzas are long. @joe suggested that we consider provider.d directory structure and automatically include provider stanzas.

**Comment by davidewatson** [comment]
Blocked on #276.

**Comment by coffeepac** [comment]
in the case of multiple clusters and multiple providers do we want to validate all providers as supported first and then start rendering files?  or is it okay to blindly create clusters until we hit an unsupported provider?

I'm leaning towards the second option being okay for now if we file an issue to fix it as we aren't publicising our ability to support multiple clusters at once but this would be a required feature if we were telling people you can control many clusters from one file.

**Comment by coffeepac** [comment]
I have a hit stopping point for tonight:  
- I have converted the aws tasks
- I am stuck on how to loop over the provider roles.  there is a new to ansible 2.2 module, [include_role](https://docs.ansible.com/ansible/include_role_module.html) which looks like it should do the trick but I'm having issues getting the loop variable to be exported in the included role

**Comment by coffeepac** [comment]
I was having indentation issues last night, got 'em sorted now.  I have migrated the aws provider, tested the broken provider and will now spin up an aws cluster.  while that runs I'll start migrating the gke templates.

**Comment by coffeepac** [comment]
Oh man, porting over the gke provider is huge.  Its a sea of kraken_config vals.  Bleargh.

**Comment by coffeepac** [comment]
got most of the gke provider role converted.  now I need to convert the ssh role as that's also keyed off of provider.

**Comment by coffeepac** [comment]
PR is up and ready to be merged after a successful review.  This should be closed when #332 is done.

**Comment by coffeepac** [comment]
PR is merged, closing.
